library/MedEx/API.php

```diff
@@ -1314,24 +1324,82 @@ public function calculateEvents($event, $start_date, $stop_date)
                                 }
                             }
                         }
-
+    
                         if ($excluded == false) {
-                            $event['pc_eventDate'] = $occurence;
-                            $event['pc_endDate'] = '0000-00-00';
-                            $events2[] = $event;
-                            $data[] = $event['pc_eventDate'];
+                            $data[] = $occurence;
                         }
+                        $occurence =& $this->MedEx->events->__increment($nd, $nm, $ny, $rfreq, $rtype);
+                        list($ny,$nm,$nd) = explode('-', $occurence);
                     }
-
-                    $occurenceYmX = date('Y-m-d', mktime(0, 0, 0, $nm+$rfreq, $nd, $ny));
-                    list($ny,$nm,$nd) = explode('-', $occurenceYmX);
-                    $occurenceYm = "$ny-$nm";
-                }
-                break;
-        }
+                    break;
+    
+                    case '2':
+                    $event_recurrspec = @unserialize($event['pc_recurrspec']);
```

Comment:
revert above line change or is a big security vulnerability (let me know if not working with the change)